app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -61,6 +79,60 @@ export default ({ commit, node }) => {
         })
       })
     },
+    async queryTxInfo({ state, dispatch }, height) {
+      let blockTxInfo = state.blockTxs.find(
+        b => b.length && b[0].height === height
+      )
+      if (blockTxInfo) {
+        return blockTxInfo
+      }
+      try {
+        blockTxInfo = await dispatch("getTxs", {
+          key: 0,
+          len:
+            state.block && state.block.data && state.block.data.txs
```

Comment:
What's the purpose of these chains of &&'s? Seems like it makes things harder to debug because if the format changes or data is missing, it will silently pass this call and have undefined behavior inside getTxs.

Reply:
this was to allow the tests that had very minimal dummy blocks to pass. but you're right if the block is malformed it should throw an error. i've updated the other tests to use a well formed block dummy